icon.schema.json

```
@@ -21,7 +21,26 @@
     "aliases": {
       "type": "array",
       "items": {
-        "type": "string"
+        "oneOf": [
+          {
+            "type": "string"
+          },
+          {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string"
+              },
+              "deprecated": {
+                "type": "boolean",
+                "default": false
+              },
+              "deprecationReason": {
+                "type": "string"
+              }
+            }
+          }
```

I would say we should enforce that you need to provide a deprecation reason.
And also enforce that you only can set a depracation reason when you set deprecated to true.
```suggestion
          {
            "type": "object",
            "required": [
              "name",
              "deprecate",
              "deprecationReason"
            ],
            "properties": {
              "name": {
                "type": "string"
              },
              "deprecated": {
                "const": true
              },
              "deprecationReason": {
                "type": "string"
              }
            }
          }
```

---

> I would say we should enforce that you need to provide a deprecation reason. And also enforce that you only can set a depracation reason when you set deprecated to true.

Good idea, although I think a pre commit hook if more feasible than enforcing it via JSON schema.

---

We can just run ajv on commit tho, we already do the lint run in the pr workflow.

---

Neato, I didn't realise we did, I've added a `dependentRequired` clause to both.
```json
  "dependentRequired": {
    "deprecated": ["deprecationReason"]
  }
```